Can't build the debug image for cross platform compling

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Version

equal or higher than v1.17.3 and lower than v1.18.0

### What happened?

When doing the corss platform compiling, if I complie from arm linux(Lima on macbook) to x86 `ARCH=amd64 NOOPT=1 NOSTRIP=1 make dev-docker-image-debug` I will get the following error.

```
 => ERROR [debug 4/5] COPY --from=builder /tmp/debug/linux/amd64/ /usr/lib/debug/                                                                                                           0.0s
------
 > [debug 4/5] COPY --from=builder /tmp/debug/linux/amd64/ /usr/lib/debug/:
------
Dockerfile:132
--------------------
 130 |
 131 |     # Copy the debug symbols across in case the binaries were stripped
 132 | >>> COPY --from=builder /tmp/debug/${TARGETOS}/${TARGETARCH}/ /usr/lib/debug/
 133 |
 134 |     # Ensure dlv finds the debug symbols. Due to CGO_ENABLED=0, we have no GNU build-id, so Delve's default search path
--------------------
ERROR: failed to solve: failed to compute cache key: failed to calculate checksum of ref bdb742e7-eafe-4220-a70e-bc133d8dad9a::bs3fwgplvs0k6wyhrn6yuqsk3: "/tmp/debug/linux/amd64": not found
make: *** [Makefile.docker:129: dev-docker-image-debug] Error 1
```



### How can we reproduce the issue?

pulling the main and cross complie with debug

### Cilium Version

latest

### Kernel Version

na

### Kubernetes Version

na

### Regression

_No response_

### Sysdump

_No response_

### Relevant log output

```shell

```

### Anything else?

_No response_

### Cilium Users Document

- [ ] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

--- a/images/cilium/Dockerfile
+++ b/images/cilium/Dockerfile
@@ -57,15 +57,20 @@ RUN --mount=type=bind,readwrite,target=/go/src/github.com/cilium/cilium \
     mv hubble /tmp/hubble/${TARGETOS}/${TARGETARCH}/hubble
 
 # Extract debug symbols to /tmp/debug and strip the binaries if NOSTRIP is not set.
+# Use the appropriate objcopy for the target architecture.
 RUN set -xe && \
     export D=/tmp/debug/${TARGETOS}/${TARGETARCH} && \
     cd /tmp/install/${TARGETOS}/${TARGETARCH} && \
     find . -type f \
       -executable \
       -exec sh -c \
-        'filename=$(basename ${0}) && \
-         objcopy --only-keep-debug ${0} ${0}.debug && \
-         if ! echo "$MODIFIERS" | grep "NOSTRIP=1" ; then objcopy --strip-all ${0} && (cd $(dirname ${0}) && objcopy --add-gnu-debuglink=${filename}.debug ${filename}) ; fi && \
+        'OBJCOPY_CMD=objcopy; \
+         if [ "${TARGETARCH}" = "amd64" ]; then OBJCOPY_CMD=x86_64-linux-gnu-objcopy; \
+         elif [ "${TARGETARCH}" = "arm64" ]; then OBJCOPY_CMD=aarch64-linux-gnu-objcopy; \
+         fi; \
+         filename=$(basename ${0}) && \
+         ${OBJCOPY_CMD} --only-keep-debug ${0} ${0}.debug && \
+         if ! echo "$MODIFIERS" | grep "NOSTRIP=1" ; then ${OBJCOPY_CMD} --strip-all ${0} && (cd $(dirname ${0}) && ${OBJCOPY_CMD} --add-gnu-debuglink=${filename}.debug ${filename}) ; fi && \
          mkdir -p $(dirname ${D}/${0}) && \
          mv -v ${0}.debug ${D}/${0}.debug' \
       {} \;

